packages/promise-cache/src/promiseCache.spec.ts

```diff
@@ -43,4 +49,72 @@ describe('PromiseCache', () => {
       errorMessage
     )
   })
+
+  it('should respect custom ttl if provided', async () => {
+    const localTTL = 0.5 // 0.5 second TTL
+
+    mockDelegate.mockResolvedValue(42)
+    await cache.wrap('testKey', mockDelegate, localTTL)
+
+    // Wait for the custom TTL to expire
+    await new Promise((resolve) => setTimeout(resolve, 600))
```

Comment:
These timeouts will make the tests take much longer than necessary. We should instead use `vi.useFakeTimers()`. This makes the tests complete in ~10 ms instead of ~4 s.

```typescript
// Setup timers at the start of the file
beforeAll(() => {
  vi.useFakeTimers()
})

// Reset timers after tests
afterAll(() => {
  vi.useRealTimers()
})

// Example tests with `vi.runAllTimers()` instead of promise resolve
it('should respect custom ttl if provided', async () => {
  const localTTL = 0.5 // 0.5 second TTL

  mockDelegate.mockResolvedValue(42)
  await cache.wrap('testKey', mockDelegate, localTTL)

  // Wait for the custom TTL to expire
  vi.runAllTimers()

  // Call again with the same key, should call mockDelegate again due to expired custom TTL
  await cache.wrap('testKey', mockDelegate, localTTL)
  expect(mockDelegate).toHaveBeenCalledTimes(2)
})
```

In the tests where we have two different resolves and assertions, we can use `vi.advanceTmersByTime` or `vi.advanceTimersToNextTimer()`:

```typescript
vi.advanceTimersByTime(600)
expect(cache.size()).toBe(1)

vi.advanceTimersByTime(500)
expect(cache.size()).toBe(0)

// OR...
vi.advanceTimersToNextTimer()
expect(cache.size()).toBe(1)

vi.advanceTimersToNextTimer()
expect(cache.size()).toBe(0)
```

Reply:
Very nice idea, I have updated the tests.